cdap-ui/app/directives/widget-container/widget-js-editor/widget-js-editor.js

```diff
@@ -34,10 +43,22 @@ angular.module(PKG.name + '.commons')
           useWrapMode: true,
           newLineMode: 'unix',
           advanced: {
-            tabSize: 2
+            tabSize: 2,
           }
         };
-
+        let height;
+        // textarea widget
+        if ($scope.mode && $scope.mode === 'plain_text') {
+          height = DEFAULT_TEXTAREA_HEIGHT;
+          if ($scope.rows) {
+            let computedHeight = $scope.rows * LINE_HEIGHT;
+            height = Math.max(computedHeight, height);
```

Comment:
I don't think we should do a `Math.max` here. If the user already specified the number of rows, we should honor that regardless, except perhaps negative number or `0` (which we should validate against).